Bravely revert "CI: Disable some Cirrus CI jobs during RIIR transition"

zanchey opened the pull request:
Test this actually works on Cirrus

krobelus commented on .cirrus.yml:
```diff
@@ -49,8 +49,7 @@ linux_arm_task:
         - ninja fish_run_tests
-    # CI task disabled during RIIR transition
-    only_if: false && $CIRRUS_REPO_OWNER == 'fish-shell'
+    only_if: $CIRRUS_REPO_OWNER == 'fish-shell'
```
`focal-arm64` fails with `cmake: command not found`

Perhaps someone published new docker images since 08b03a733ab (docker_run_tests.sh: stop using cmake, 2025-06-19) was done. Anyway, I think we can fix this by calling `build_tools/check.sh` here, no need for CMake.
Note that `FISH_CHECK_LINT=false` is already set in the dockerfile.

Last time I updated the Linux builds above, I needed to update something in the docker images, which meant that I needed to publish them to ghcr.io. I haven't written down the steps but it should be done with `docker build` (as done by `docker/docker_run_tests.sh`) and `docker pull`, though these might require an ARM machine

zanchey replied:
Yeah, I have a workflow in draft to make new builds.

krobelus replied:
So Linux builds are failing because `FISH_CHECK_LINT` wasn't set,
meaning that `check.sh` will run all warnings/formatters etc.

I'm running tests right now with `FISH_CHECK_LINT=false` set explicit:
https://github.com/fish-shell/fish-shell/tree/cirrus-ci
that should fix it.
(Of course we could check warnings on these platforms as well in future, but it's not very important).

So it looks like docker images will automatically be pushed to the registry whenever changes to `docker/` are pushed to master.

I assume they're already published (via https://github.com/fish-shell/fish-shell/actions/runs/18267743492, looks like the push of f6d93f2fdb0 (GitHub Actions: add workflow to build Docker images for CI, 2025-10-06) triggered it, which is surprising but it is what we want).

so I'll change the image paths back from my temporary ones:

```
diff --git a/.cirrus.yml b/.cirrus.yml
index 583d16ebd1..9cacfb4bdc 100644
--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -6,20 +6,20 @@
     matrix:
         - name: alpine
           container: &step
-              image: ghcr.io/krobelus/fish-ci/alpine:latest
+              image: ghcr.io/fish-shell/fish-ci/alpine:latest
               memory: 4GB
```

zanchey committed:
Bravely revert "CI: Disable some Cirrus CI jobs during RIIR transition"

This reverts what remains of commit
91be7489bc71a0268996227ddd733c78f898e703.

Closes #11871
Closes #11884

krobelus committed:
cirrus.yml: switch back to org docker image paths

These should work now that we have (automatically-updated) docker
builds via .github/workflows/docker_builds.yml.

Ref: https://github.com/fish-shell/fish-shell/pull/11884#discussion_r2405536855

krobelus commented on .cirrus.yml:
```diff
@@ -42,8 +42,7 @@ linux_arm_task:
         - FISH_TEST_MAX_CONCURRENCY=6 build_tools/check.sh
-    # CI task disabled during RIIR transition
-    only_if: false && $CIRRUS_REPO_OWNER == 'fish-shell'
+    only_if: $CIRRUS_REPO_OWNER == 'fish-shell'
```
I think we want something like `only_if: repo owner is fish-shell` also for
`.github/workflows/docker_builds.yml`,
else these might run (and typically fail) whenever someone pushes to their github fork's `master` branch
(there's usually not too much need for anyone to do that, but people do and it does make some things easier for the lazy
(by default github workflows only run from master, unless `gh workflow run --ref` is used etc)

krobelus commented:
TODO: need to either re-revert b1e8fdfaa2c (Revert "CI: use build_tools/check.sh in Cirrus CI", 2025-10-06),
or keep using cmake (I have not had time to figure out what was the problem; some of them symptoms are documented in the commits I pushed to this PR).

The fact that `.github/workflows/docker_builds.yml` only writes to a central registry that's directly used by `master` is not ideal;
maybe we can set things up so it's easy to test everything in a temporary branch, e.g. by allowing `workflow_dispatch` triggers to use a custom docker registry namespace

zanchey commented:
> The fact that `.github/workflows/docker_builds.yml` only writes to a central registry that's directly used by `master` is not ideal; maybe we can set things up so it's easy to test everything in a temporary branch, e.g. by allowing `workflow_dispatch` triggers to use a custom docker registry namespace

You can test it in your own fork and it will write to your own package registry. A dispatch trigger would be a fine thing to add.

I was going to poke at this some more but it looks like you want to take it over?

krobelus commented:
> You can test it in your own fork and it will write to your own package registry

right, that should work (probably some of the ones in https://github.com/krobelus/fish-shell/actions/runs/18277577848 failed because I had never pushed those images before, so I get a permissions error).

> I was going to poke at this some more but it looks like you want to take it over?

Feel free to take it back (and drop the bad commit that's unrelated to Cirrus).
We could stick with cmake for now, to keep it close to current GitHub actions, but `check.sh` should work, and might be easier because I already removed CMake from the docker images

krobelus committed:
cirrus: disable jammy-armv7-32bit for now

This fails with "exec format error" because our container is built
on a 64 bit system on GitHub Actions.  Not yet sure how to fix that.

krobelus committed:
WIP cirrus: trigger after github action finishes

(TODO this is totally untested and mostly vibe coded.)

Whenever we push changes do docker/**, our docker images for Cirrus
CI will be rebuilt.

However, the Cirrus CI jobs will kick off at the same time as the
Docker builds, so they will likely use old images.  This can cause
surprising (albeit transient) failures.

Fix this by having Cirrus wait for GitHub Actions.

This addresses the second part of
https://github.com/fish-shell/fish-shell/pull/11884#discussion_r2423344925
